src/com/amazon/ion/util/Equivalence.java

```diff
@@ -115,7 +129,77 @@
      */
     private static final boolean PUBLIC_COMPARISON_API = false;
 
-    private Equivalence() {
+    /**
+     * Configuration that requires strict data equivalence.
```

Comment:
Glancing through the existing code and this PR, I wasn't able to find a place where 'strict' data equivalence was defined in user-facing documentation. Is there such a place? If not, I think we should add it to the class docs for `Equivalence` and on the relevant methods for the Builder.

Reply:
I think it's sufficiently defined on `Builder.withStrict` and `Equivalence.ionEqualsByContent`.

Reply:
The docs you refer to use the phrase 'strict data equivalence' without ever defining it:

**`Builder.withStrict`**
```
When true, checks for strict data equivalence over two Ion Values.
When false, checks for structural data equivalence over two Ion Values.
```

It's not clear to me what that means, apart from being distinct from 'structural data equivalence'.

**`Equivalence.ionEqualsByContent`**
```
Checks for structural data equivalence over two Ion Values. 
That is, equivalence without considering any annotations.
```

This would seem to suggest that 'strict' equivalence boils down to considering annotations. Is that accurate?

Reply:
Fair enough. Ion data model equivalence can't be described succinctly enough for a Javadoc comment, but I will refer to the Ion specification.